pom.xml

```diff
@@ -153,6 +155,30 @@
 				<artifactId>appengine-maven-plugin</artifactId>
 				<version>${appengine-maven-plugin-version}</version>
 			</plugin>
+			<plugin>
+				<groupId>pl.project13.maven</groupId>
+				<artifactId>git-commit-id-plugin</artifactId>
+				<version>4.0.0</version>
+				<executions>
+					<execution>
+						<id>get-the-git-infos</id>
+						<goals>
+							<goal>revision</goal>
+						</goals>
+						<phase>initialize</phase>
+					</execution>
+				</executions>
+				<configuration>
+					<dateFormat>yyyyMMdd-HHmm</dateFormat>
+					<generateGitPropertiesFile>true</generateGitPropertiesFile>
+					<generateGitPropertiesFilename>${project.build.outputDirectory}/git.properties</generateGitPropertiesFilename>
+					<includeOnlyProperties>
+						<includeOnlyProperty>^git.commit.id.describe-short$</includeOnlyProperty>
```

Comment:
hmm, when built I only see git.commit.time in the git.properties file. Where's the git.commit.id.describe-short?
```
➜  cat target/SynapseAWSConsoleLogin/WEB-INF/classes/git.properties
#Generated by Git-Commit-Id-Plugin
#Thu May 28 08:27:42 PDT 2020
git.commit.time=20200527-1804
```

Reply:
I had it along with 'describe' and it disappeared when I took 'describe' out... I'm just going to put it back in, makes no sense but... With a named tag (and uncommited changes):
 
#Generated by Git-Commit-Id-Plugin
#Thu May 28 12:21:50 PDT 2020
git.commit.id.describe=0.31-dirty
git.commit.id.describe-short=0.31-dirty
git.commit.time=20200527-1804

Reply:
what is `0.31-dirty`?  I was expecting a git SHA.  Could we add one of the properties that will give us the git SHA?

Reply:
We're going full circle, 'describe' will use a tag if there's one (ot that it's not a bad idea if we put a annotated tag before release). The original 'abbrev' still shows me the first chars of the SHA:
#Generated by Git-Commit-Id-Plugin
#Fri May 29 09:41:30 PDT 2020
git.commit.id.abbrev=8eda288
git.commit.id.describe=0.31-3-g8eda288-dirty
git.commit.id.describe-short=0.31-3-dirty
git.commit.time=20200529-0939
Go back to that?
There's also some wonkiness around the OutputDirectory since I removed the name override, the file gets generated but does not get included in the .war. Latest pull should fix that, though I don't understand why I can't just delete the OutputDirectory altogether...

Reply:
what we really want is the `abbrev` and `dirty`.  I think describe is optional.  you can keep it if you think it's useful.